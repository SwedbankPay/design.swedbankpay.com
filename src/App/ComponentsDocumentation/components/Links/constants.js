import React from "react";
import Link  from "@components/Link";

export const overviewLinks = {
    id: "overviewLinks",
    tabsId: "overviewLinksTabs",
    elements: [
        {
            tab: "Inline",
            component: <>
                <p>Paragraph text with a <Link linkText="Inline link"/></p></>,
            options: {
                checkbox: [
                    {
                        inputs: [
                            {
                                id: "external_link",
                                name: "External link",
                                value: {
                                    value: "<p><p/>//TODO"
                                }
                            }
                        ]
                    }
                ],
                radio: [
                    {
                        id: "font_size_radio",
                        title: "Font size",
                        values: [
                            {
                                name: "Normal",
                                value: {
                                    size: "lg"
                                }
                            },
                            {
                                name: "Small",
                                value: {
                                    size: "sm"
                                }
                            }
                        ]
                    }
                ]
            },
            title: "Inline links",
            description: "Inline links appear within paragraphs or sentences. They’re used to direct users to content that’s not directly related to the journey they’re on. If a link is directing the user to an external web page, this should be indicated using an icon and the page should open in a new window. The font size should match the font size of the text it is inline with."
        },
        {
            tab: "Standalone",
            component: <Link linkText="Standalone link"/>,
            options: {
                checkbox: [
                    {
                        inputs: [
                            {
                                id: "icon_check",
                                name: "With icon",
                                value: {
                                    icon: "cloud"
                                }
                            }
                        ]
                    }
                ],
                radio: [
                    {
                        id: "font_size_radio",
                        title: "Font size",
                        values: [
                            {
                                name: "Normal",
                                value: {
                                    size: "lg"
                                }
                            },
                            {
                                name: "Small",
                                value: {
                                    size: "sm"
                                }
                            }
                        ]
                    }
                ]
            },
            title: "Standalone links",
            description: "Standalone links are separate to paragraph content and appear on there own. They can be user with a leading icon to further convey the result of clicking the link. Links with icons are not underlined."
        }
    ]
};
