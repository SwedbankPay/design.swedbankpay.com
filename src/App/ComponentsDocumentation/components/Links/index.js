import React from "react";

import { ComponentPreview, DocContainer } from "@docutils";
import ActionLinkComponent from "@components/ActionLink";
import LinkComponent from "@components/Link";
import ButtonComponent from "@components/Button";
import CodeTags from "@components/CodeTags";

const TextLinks = () => (
    <>
        <h2 id="text-links">Text links</h2>
        <p>Text links are links used in paragraphs of text.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <p>{"\n"}
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.{"\n"}
                Ut enim ad minim veniam, quis nostrud <LinkComponent linkText="exercitation" /> ullamco laboris nisi ut aliquip ex ea commodo consequat.{"\n"}
            </p>
        </ComponentPreview>

        <h3>Small link</h3>
        <p>Use links in small with legal texts, such as links to terms and contitions</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <small>{"\n"}
                Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.{"\n"}
                Ut enim ad minim veniam, quis nostrud <LinkComponent linkText="exercitation" /> ullamco laboris nisi ut aliquip ex ea commodo consequat.{"\n"}
            </small>
        </ComponentPreview>
    </>
);

const ButtonLinks = () => (
    <>
        <h2 id="button-links">Links with buttons</h2>
        <p>
            To use links with buttons, for instance as a cancel button,
            use <CodeTags type="secondary" code=".btn.btn-link" /> in a <CodeTags type="primary" code="<button>" />.
        </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <ButtonComponent type="primary" label="Approve" />{"\n"}
            <ButtonComponent type="link" label="Cancel"/>
        </ComponentPreview>
    </>
);

const IconLinks = () => (
    <>
        <h2 id="icon-links">Links with icons</h2>
        <p>
            Use class <CodeTags type="secondary" code=".icon-link" /> on links with icons. Using icon with links can help further
            convey the result of clicking on the link, as can be seen in the examples below.
        </p>
        <h4>Navigates to a previous page</h4>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <LinkComponent linkText="My invoices" leftIcon="arrow_back_ios"/>
        </ComponentPreview>
        <h4>Enables editing</h4>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <LinkComponent linkText="My invoices" leftIcon="edit"/>
        </ComponentPreview>
        <h4>Opens in a new tab</h4>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <LinkComponent linkText="My invoices" rightIcon="open_in_new"/>
        </ComponentPreview>
    </>
);

const ActionLink = () => (
    <>
        <h2 id="action-link">Action link</h2>
        <p>Action links.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <ActionLinkComponent linkText="My link"/>
        </ComponentPreview>
    </>
);

const badge = {
    type: "badge-default badge-number",
    text: "40"
};

const ActionLinkBadge = () => (
    <>
        <h2 id="action-link-badge">Action link with Badge</h2>
        <p>Action links.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <ActionLinkComponent linkText="My link" badge={badge}/>
        </ComponentPreview>
    </>
);

const ActionLinkSmallText = () => (
    <>
        <h2 id="action-link-small-text">Action link with small text</h2>
        <p>A small text can be displayed as a part of the Action Link. There are two types of Action Links with small text, one with the inline small text and one with the texts on different lines.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <ActionLinkComponent linkText="My link" smallText="This link directs to my page"/>
        </ComponentPreview>
        <p>Use <CodeTags type="secondary" code=".action-link-multiline"/> for the Action Link with the texts on different lines.</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <ActionLinkComponent linkText="My link" smallText="This link directs to my page" multiline={true} />
        </ComponentPreview>
    </>
);

const ActionLinkBadgeSmallText = () => (
    <>
        <h2 id="action-link-badge-small-text">Action link with Badge and small text</h2>
        <p>Action Links can contain both Badge and small text</p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <ActionLinkComponent linkText="My link" smallText="This link directs to my page" badge={badge}/>
        </ComponentPreview>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <ActionLinkComponent linkText="My link" smallText="This link directs to my page" multiline={true} badge={badge}/>
        </ComponentPreview>
    </>
);

const ActionLinkNewTab = () => (
    <>
        <h2 id="action-link-new-tab">Action link that opens a new tab</h2>
        <p>An Action Link used in cases where a new tab is supposed to open is provided. Use <CodeTags type="secondary" code=".action-link-new-tab" /> when it is known that the target will open in a new tab. </p>
        <ComponentPreview language="html" showCasePanel codeFigure>
            <ActionLinkComponent linkText="My link" newTab={true} />
        </ComponentPreview>
    </>
);

const Links = () => (
    <DocContainer docToc>
        <p className="lead">
            Links
        </p>
        <TextLinks />
        <ButtonLinks />
        <IconLinks />
        <ActionLink />
        <ActionLinkBadge />
        <ActionLinkSmallText />
        <ActionLinkBadgeSmallText />
        <ActionLinkNewTab />
    </DocContainer>
);

export default Links;

/* For testing */
export { TextLinks, ButtonLinks, IconLinks, ActionLink, ActionLinkBadge, ActionLinkSmallText, ActionLinkBadgeSmallText, ActionLinkNewTab };
