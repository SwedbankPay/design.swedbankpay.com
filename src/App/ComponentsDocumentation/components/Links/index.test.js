import React from "react";
import { shallow } from "enzyme";

import Links, {
    TextLinks,
    ButtonLinks,
    IconLinks,
    ActionLink,
    ActionLinkBadge,
    ActionLinkSmallText,
    ActionLinkBadgeSmallText,
    ActionLinkNewTab
} from "./index";

describe("Components: Links", () => {
    it("is defined", () => {
        expect(Links).toBeDefined();
    });

    it("renders", () => {
        const wrapper = shallow(<Links />);

        expect(wrapper).toMatchSnapshot();
    });

    describe("TextLinks", () => {
        it("is defined", () => {
            expect(TextLinks).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<TextLinks />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("ButtonLinks", () => {
        it("is defined", () => {
            expect(ButtonLinks).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<ButtonLinks />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("IconLinks", () => {
        it("is defined", () => {
            expect(IconLinks).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<IconLinks />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("ActionLink", () => {
        it("is defined", () => {
            expect(ActionLink).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<ActionLink />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("ActionLinkBadge", () => {
        it("is defined", () => {
            expect(ActionLinkBadge).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<ActionLinkBadge />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("ActionLinkSmallText", () => {
        it("is defined", () => {
            expect(ActionLinkSmallText).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<ActionLinkSmallText />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("ActionLinkBadgeSmallText", () => {
        it("is defined", () => {
            expect(ActionLinkBadgeSmallText).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<ActionLinkBadgeSmallText />);

            expect(wrapper).toMatchSnapshot();
        });
    });

    describe("ActionLinkNewTab", () => {
        it("is defined", () => {
            expect(ActionLinkNewTab).toBeDefined();
        });

        it("renders", () => {
            const wrapper = shallow(<ActionLinkNewTab />);

            expect(wrapper).toMatchSnapshot();
        });
    });

});
