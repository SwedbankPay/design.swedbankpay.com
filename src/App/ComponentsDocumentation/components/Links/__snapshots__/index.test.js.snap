// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Components: Links ActionLink renders 1`] = `
<Fragment>
  <h2
    id="action-link"
  >
    Action link
  </h2>
  <p>
    Action links.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <ActionLink
      linkText="My link"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Links ActionLinkBadge renders 1`] = `
<Fragment>
  <h2
    id="action-link-badge"
  >
    Action link with Badge
  </h2>
  <p>
    Action links.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <ActionLink
      badge={
        Object {
          "text": "40",
          "type": "badge-default badge-number",
        }
      }
      linkText="My link"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Links ActionLinkBadgeSmallText renders 1`] = `
<Fragment>
  <h2
    id="action-link-badge-small-text"
  >
    Action link with Badge and small text
  </h2>
  <p>
    Action Links can contain both Badge and small text
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <ActionLink
      badge={
        Object {
          "text": "40",
          "type": "badge-default badge-number",
        }
      }
      linkText="My link"
      smallText="This link directs to my page"
    />
  </ComponentPreview>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <ActionLink
      badge={
        Object {
          "text": "40",
          "type": "badge-default badge-number",
        }
      }
      linkText="My link"
      multiline={true}
      smallText="This link directs to my page"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Links ActionLinkNewTab renders 1`] = `
<Fragment>
  <h2
    id="action-link-new-tab"
  >
    Action link that opens a new tab
  </h2>
  <p>
    An Action Link used in cases where a new tab is supposed to open is provided. Use 
    <CodeTags
      code=".action-link-new-tab"
      type="secondary"
    />
     when it is known that the target will open in a new tab. 
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <ActionLink
      linkText="My link"
      newTab={true}
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Links ActionLinkSmallText renders 1`] = `
<Fragment>
  <h2
    id="action-link-small-text"
  >
    Action link with small text
  </h2>
  <p>
    A small text can be displayed as a part of the Action Link. There are two types of Action Links with small text, one with the inline small text and one with the texts on different lines.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <ActionLink
      linkText="My link"
      smallText="This link directs to my page"
    />
  </ComponentPreview>
  <p>
    Use 
    <CodeTags
      code=".action-link-multiline"
      type="secondary"
    />
     for the Action Link with the texts on different lines.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <ActionLink
      linkText="My link"
      multiline={true}
      smallText="This link directs to my page"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Links ButtonLinks renders 1`] = `
<Fragment>
  <h2
    id="button-links"
  >
    Links with buttons
  </h2>
  <p>
    To use links with buttons, for instance as a cancel button, use 
    <CodeTags
      code=".btn.btn-link"
      type="secondary"
    />
     in a 
    <CodeTags
      code="<button>"
      type="primary"
    />
    .
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <Button
      label="Approve"
      type="primary"
    />
    

    <Button
      label="Cancel"
      type="link"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Links IconLinks renders 1`] = `
<Fragment>
  <h2
    id="icon-links"
  >
    Links with icons
  </h2>
  <p>
    Use class 
    <CodeTags
      code=".icon-link"
      type="secondary"
    />
     on links with icons. Using icon with links can help further convey the result of clicking on the link, as can be seen in the examples below.
  </p>
  <h4>
    Navigates to a previous page
  </h4>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <Link
      leftIcon="arrow_back_ios"
      linkText="My invoices"
    />
  </ComponentPreview>
  <h4>
    Enables editing
  </h4>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <Link
      leftIcon="edit"
      linkText="My invoices"
    />
  </ComponentPreview>
  <h4>
    Opens in a new tab
  </h4>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <Link
      linkText="My invoices"
      rightIcon="open_in_new"
    />
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Links TextLinks renders 1`] = `
<Fragment>
  <h2
    id="text-links"
  >
    Text links
  </h2>
  <p>
    Text links are links used in paragraphs of text.
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <p>
      

      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
      

      Ut enim ad minim veniam, quis nostrud 
      <Link
        linkText="exercitation"
      />
       ullamco laboris nisi ut aliquip ex ea commodo consequat.
      

    </p>
  </ComponentPreview>
  <h3>
    Small link
  </h3>
  <p>
    Use links in small with legal texts, such as links to terms and contitions
  </p>
  <ComponentPreview
    codeFigure={true}
    language="html"
    showCasePanel={true}
  >
    <small>
      

      Lorem ipsum dolor sit amet, consectetur adipiscing elit, sed do eiusmod tempor incididunt ut labore et dolore magna aliqua.
      

      Ut enim ad minim veniam, quis nostrud 
      <Link
        linkText="exercitation"
      />
       ullamco laboris nisi ut aliquip ex ea commodo consequat.
      

    </small>
  </ComponentPreview>
</Fragment>
`;

exports[`Components: Links renders 1`] = `
<DocContainer
  docToc={true}
>
  <p
    className="lead"
  >
    Links
  </p>
  <TextLinks />
  <ButtonLinks />
  <IconLinks />
  <ActionLink />
  <ActionLinkBadge />
  <ActionLinkSmallText />
  <ActionLinkBadgeSmallText />
  <ActionLinkNewTab />
</DocContainer>
`;
